Documentation/design/initphase.md

```diff
@@ -0,0 +1,44 @@
+# Init Phase
+
+## Goals
+
+1. Generates `InstallConfig` by prompting user for required parameters, and populating the rest of configuration with default values.
+2. Make useful decisions on default values for most options.
+3. Init phase should never prompt the user for more than 10 options unless there is a very strong reason.
```

Comment:
nit: Seems somewhat arbitrary to say 10 (and I'd hope we get nowhere near that). The intention is that `init` should only prompt for things it **cannot** default (e.g. for AWS, `basedomain`).